cranelift/codegen/src/isa/s390x/lower.isle

```diff
@@ -3842,46 +3842,6 @@
 (rule (lower (debugtrap))
       (side_effect (debugtrap_impl)))
 
-
-;;;; Rules for `trapif` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-
-;; Similarly to `selectif_spectre_guard`, we only recognize specific patterns
-;; generated by common code here.  Others will fail to lower.
-
-;; Recognize the case of `ifcmp` feeding into `trapif`.  Directly generate
-;; the desired comparison here; there is no separate `ifcmp` lowering.
-
-(rule (lower (trapif int_cc (ifcmp x y) trap_code))
-      (side_effect (trap_if_bool (icmp_val $false int_cc x y) trap_code)))
-
-;; Recognize the case of `iadd_ifcout` feeding into `trapif`.  Note that
-;; in the case, the `iadd_ifcout` is generated by a separate lowering
-;; (in order to properly handle the register output of that instruction.)
-;;
-;; The flags must not have been clobbered by any other instruction between the
-;; iadd_ifcout and this instruction, as verified by the CLIF validator; so we
-;; can simply rely on the condition code here.
-;;
-;; IaddIfcout is implemented via a ADD LOGICAL instruction, which sets the
```

Comment:
The part of the comment starting here should still be kept - the new `uadd_overflow_trap` still uses the condition code mask described here.   The comment should probably just be moved there.